packages/authorization-process/test/api/addClientPurpose.test.ts

```diff
@@ -87,107 +93,87 @@ describe("API /clients/{clientId}/purposes authorization test", () => {
-    expect(res.status).toBe(400);
-  });
-
-  it("Should return 400 if passed an invalid field", async () => {
-    const token = generateToken(authRole.ADMIN_ROLE);
-    const res = await makeRequest(token, "invalid");
-    expect(res.status).toBe(400);
-  });
-
-  it("Should return 409 for purposeAlreadyLinkedToClient", async () => {
-    authorizationService.addClientPurpose = vi
-      .fn()
-      .mockRejectedValue(
-        purposeAlreadyLinkedToClient(mockPurpose.id, mockClient.id)
-      );
-    const token = generateToken(authRole.ADMIN_ROLE);
-    const res = await makeRequest(token, mockClient.id);
-    expect(res.status).toBe(409);
-  });
-
-  it("Should return 403 for clientKindNotAllowed", async () => {
-    authorizationService.addClientPurpose = vi
-      .fn()
-      .mockRejectedValue(clientKindNotAllowed(mockClient.id));
-    const token = generateToken(authRole.ADMIN_ROLE);
-    const res = await makeRequest(token, mockClient.id);
-    expect(res.status).toBe(403);
-  });
-
-  it("Should return 403 for organizationNotAllowedOnClient", async () => {
-    authorizationService.addClientPurpose = vi
-      .fn()
-      .mockRejectedValue(
-        organizationNotAllowedOnClient(generateId(), mockClient.id)
-      );
-    const token = generateToken(authRole.ADMIN_ROLE);
-    const res = await makeRequest(token, mockClient.id);
-    expect(res.status).toBe(403);
-  });
+  it.each([
+    {
+      name: "clientNotFound",
+      error: clientNotFound(mockClient.id),
+      expectedStatus: 404,
+    },
+    {
+      name: "purposeNotFound",
+      error: purposeNotFound(mockPurpose.id),
+      expectedStatus: 404,
+    },
+    {
+      name: "noAgreementFoundInRequiredState",
+      error: noAgreementFoundInRequiredState(mockEservice.id, mockConsumerId),
+      expectedStatus: 400,
+    },
+    {
+      name: "noPurposeVersionsFoundInRequiredState",
+      error: noPurposeVersionsFoundInRequiredState(mockPurpose.id),
+      expectedStatus: 400,
+    },
+    {
+      name: "eserviceNotDelegableForClientAccess",
+      error: eserviceNotDelegableForClientAccess(mockEservice),
+      expectedStatus: 400,
+    },
+    {
+      name: "purposeAlreadyLinkedToClient",
+      error: purposeAlreadyLinkedToClient(mockPurpose.id, mockClient.id),
+      expectedStatus: 409,
+    },
+    {
+      name: "clientKindNotAllowed",
+      error: clientKindNotAllowed(mockClient.id),
+      expectedStatus: 403,
+    },
+    {
+      name: "organizationNotAllowedOnClient",
+      error: organizationNotAllowedOnClient(generateId(), mockClient.id),
+      expectedStatus: 403,
+    },
+    {
+      name: "organizationNotAllowedOnPurpose",
+      error: organizationNotAllowedOnPurpose(generateId(), mockPurpose.id),
+      expectedStatus: 403,
+    },
+    {
+      name: "purposeDelegationNotFound",
+      error: purposeDelegationNotFound(generateId()),
+      expectedStatus: 500,
+    },
+  ])(
+    "Should return $expectedStatus for $name",
```

Comment:
If you want to avoid writing the `name` field in the array, this should work, since the error functions return an object with the `code` field
```suggestion
    "Should return $expectedStatus for $error.code",
```

Reply:
I tried it for my own testing purposes, and I believe the `$object.field` solution doesn't work because _Vitest_ doesn't support accessing object fields directly. [This](https://vitest.dev/api/#:~:text=You%20can%20also%20access%20object%20properties%20and%20array%20elements%20with%20%24%20prefix%3A) is the documentation I based my attempt on, but it doesn’t exactly reflect the desired behavior: